tripy/tripy/flat_ir/ops/reduce.py

```diff
@@ -71,11 +71,14 @@ def to_mlir(self, operands):
 
 
 @dataclass(repr=False)
-class ArgMinMaxOp(ReduceOp):
+class ArgMinMaxOp(BaseFlatIROp):
```

Comment:
For some reason, trace ArgMinMaxOp gets mapped to ReduceOp instead of ArgMinMaxOp intermittently.
@pranavm-nvidia Any idea why `self.to_flat_ir` gets confused between an op and its derived class?
I tried debugging since for a bit and suspect it has something to do with decorator overriding.

Reply:
Can you elaborate? Do you mean that when you call `to_flat_ir` it sometimes calls the `Reduce` implementation instead?

Reply:
Yes. I need to check if this is true without my changes as well.

Reply:
There was a bug while overriding build method. I have fixed this by ensuring we override derived class build method before base class build method.